### PR TITLE
[change] Random pause for every 10 successfull sent requests

### LINKED ISSUE #128: [change] Introduce random pause in sending of accumulated data
Introduce random pause in sending of accumulated data at the end of this loop:
https://github.com/openwisp/openwrt-openwisp-monitoring/blob/adf28d536fd61c22ca4f021233e9d8c0bbb49d06/openwisp-monitoring/files/monitoring.agent#L122

The pause should be introduced only every 10 requests, it shall be of a random amount of seconds, comprised between 1 and 5 and it shall be introduced only if the sending was successful.

### PR DESCRIPTION
**Pull request by Dhanus3133**
Closes #128

### CODE REVIEW

File: openwisp-monitoring/files/monitoring.agent [hunk 1]
```diff
@@ -195,6 +197,11 @@ send_data() {
 			done
 			# retry sending same data again in next cycle
 			[ "$failures" -eq "$MAX_RETRIES" ] && break
+            # pause for a while after every 10 successful request sent
+			if [ $((success_count % 10)) -eq 0 ]; then
```
Comment (pandafy) [inline openwisp-monitoring/files/monitoring.agent:201]: Isn't `0 % 10 == 0` ? Why not just use `[ $success_rate -ge 10 ]` which is more readable?

**Review by pandafy — CHANGES_REQUESTED** [review]

File: openwisp-monitoring/files/monitoring.agent [hunk 2]
```diff
@@ -195,6 +197,11 @@ send_data() {
 			done
 			# retry sending same data again in next cycle
 			[ "$failures" -eq "$MAX_RETRIES" ] && break
+            # pause for a while after every 10 successful request sent
```
Comment (nemesifier) [inline openwisp-monitoring/files/monitoring.agent:200]: watch out from indentation mismatches like this

File: openwisp-monitoring/files/monitoring.agent [hunk 3]
```diff
@@ -118,6 +118,7 @@ handle_sigusr1() {
 }
 
 send_data() {
+	success_count=0
```
Comment (nemesifier) [inline openwisp-monitoring/files/monitoring.agent:121]: If we start from zero, the pause will not be executed after 10 requests, but 11, right? So If we want the implementation to follow the spec and if we want the comment below to be technically correct, I think we should start from 1 here.

File: openwisp-monitoring/files/monitoring.agent (hunk 1 above)
Comment (Dhanus3133) [reply openwisp-monitoring/files/monitoring.agent:201]: `-ge` makes it run whenever the `success_rate` is more than or equal to 10. But as per the issue, it was meant for every 10 requests. Was I right on this one?

File: openwisp-monitoring/files/monitoring.agent (hunk 2 above)
Comment (Dhanus3133) [reply openwisp-monitoring/files/monitoring.agent:200]: Sorry MB, formatting issue will fix it

File: openwisp-monitoring/files/monitoring.agent (hunk 3 above)
Comment (Dhanus3133) [reply openwisp-monitoring/files/monitoring.agent:121]: MB you're right. Will update it!

File: openwisp-monitoring/files/monitoring.agent (hunk 1 above)
Comment (nemesifier) [reply openwisp-monitoring/files/monitoring.agent:201]: The requirement is to pause every 10 successful requests.
I don't see any `success_rate` variable in this diff, @pandafy where is `success_rate` coming from?
I only see `success_count` here.

File: openwisp-monitoring/files/monitoring.agent [hunk 4]
```diff
@@ -118,6 +118,7 @@ handle_sigusr1() {
 }
 
 send_data() {
+	success=1
```
Comment (nemesifier) [inline openwisp-monitoring/files/monitoring.agent:121]: After having reviewed the code I think you were right in setting this to zero, my calculation was incorrect.

**Review by nemesifier — APPROVED** [review]

**Review by nemesifier — APPROVED** [review]

**Pull request merged**